107/src/main/java/org/ehcache/jsr107/EhcacheCachingProvider.java

```diff
@@ -156,6 +156,61 @@ private Eh107CacheManager createCacheManager(URI uri, Configuration config, Prop
             new ConfigurationMerger(config, jsr107Service, cacheLoaderWriterFactory));
   }
 
+  /**
+   * Enables to create a JSR-107 {@link CacheManager} based on the provided Ehcache {@link CacheManagerBuilder}.
+   *
+   * @param uri the URI identifying this cache manager
+   * @param classLoader class loader where the cache manager will be registered
+   * @param builder the builder to create the underlying EhcacheManager
+   *
+   * @return a cache manager
+   */
+  public CacheManager getCacheManager(URI uri, ClassLoader classLoader, CacheManagerBuilder<org.ehcache.CacheManager> builder) {
+    Eh107CacheManager cacheManager;
+
+    synchronized (cacheManagers) {
```

Comment:
We can probably extract this common block between here and the existing code in to a combination of a `computeIfAbsent` and a `compute`.

Reply:
Yes. I've been purposefully lazy.